applications/TrilinosApplication/external_includes/amgcl_mpi_solver_impl.cpp

```diff
@@ -45,6 +45,11 @@ void AMGCLScalarSolve(
     bool use_gpgpu
     )
 {
+    const Epetra_Comm& r_epetra_comm = rA.Comm();
+    const Epetra_MpiComm& r_epetra_mpi_comm = dynamic_cast<const Epetra_MpiComm&>(r_epetra_comm); // cannot use static_cast due to virtual inheritance
+
+    MPI_Comm the_comm = r_epetra_mpi_comm.Comm();
```

Comment:
Looks like `Epetra_Comm` inherits `Epetra_MpiComm`. If that is indeed the case, would not the following work?
```cpp
MPI_Comm the_comm = ra.Comm().Comm();
```

Reply:
Sorry, it seems [the other way around](https://docs.trilinos.org/dev/packages/epetra/doc/html/classEpetra__Comm.html#details) (`Epetra_MpiComm` inherits `Epetra_Comm`).

Reply:
yes exactly thats why this doesn't work

In Kratos we have a fct for this:
https://github.com/KratosMultiphysics/Kratos/blob/b947b82c90dfcbf13d60511427f85990d36b90be/kratos/mpi/sources/mpi_data_communicator.cpp#L399-L410

Unfortunately I haven't found any equivalent in Trilinos

Reply:
I asked in trilinos but no better answer:
https://github.com/trilinos/Trilinos/issues/10122